p2p/src/peer.rs

```diff
@@ -258,11 +258,40 @@ impl Peer {
 		}
 	}
 
+	pub fn send_tx_kernel_hash(&self, h: Hash) -> Result<bool, Error> {
+		if !self.tracking_adapter.has(h) {
+			debug!("Send tx kernel hash {} to {}", h, self.info.addr);
+			self.connection
+				.as_ref()
+				.unwrap()
+				.send(h, msg::Type::TransactionKernel)?;
+			Ok(true)
+		} else {
+			debug!(
+				"Not sending tx kernel hash {} to {} (already seen)",
+				h, self.info.addr
+			);
+			Ok(false)
+		}
+	}
+
 	/// Sends the provided transaction to the remote peer. The request may be
 	/// dropped if the remote peer is known to already have the transaction.
+	/// We support broadcast of lightweight tx kernel hash
+	/// so track known txs by kernel hash.
 	pub fn send_transaction(&self, tx: &core::Transaction) -> Result<bool, Error> {
-		if !self.tracking_adapter.has(tx.hash()) {
-			debug!("Send tx {} to {}", tx.hash(), self.info.addr);
+		let kernel = &tx.kernels()[0];
```

Comment:
What happens if a transaction has multiple kernels? If a node receiving this message already has only the first kernel in their pool, it won't request the full trasaction and miss out on the other kernels (which it might not all have)

Reply:
That's a good point. We were originally thinking of doing this by sending the full set of kernels but thought this was overly complex.

In normal operation we will not see aggregate txs like this because aggregation _should_ only occur during the Dandelion stem phase.

But there is nothing to stop a node from arbitrarily (maliciously or otherwise) aggregating transactions during regular tx propagation.

Say a node aggregates `tx1 (k1)` and `tx2 (k2)` to give `tx3 (k1, k2)`.
Now we have two different scenarios depending on _which_ kernel is chosen here - 
  * if `k1` then `tx3` will be ignored by nodes that are aware of `tx1`
  * if `k2` then `tx3` will be ignored by nodes that are aware of `tx2`

@ignopeverell thoughts?

Today we make an effort to "de-aggregate" aggregated txs like this - 
  * if I see `tx3` after `tx1` then I will recreate `tx2` (the difference) and broadcast `tx2`

This PR will reduce the effectiveness of this significantly.

Reply:
That said - the result of this is a node may send a "kernel hash" of an aggregated tx, only for it to be ignored by all its peers because they have seen this kernel before (and mistakenly think they have seen the full tx before).

Is this any different to a node simply not notifying its peers of the new tx?

To attempt to cause mischief with this it would be easier and cheaper just to drop txs and not notify anyone.